content/en/docs/reference/access-authn-authz/authentication.md

```diff
@@ -735,11 +735,16 @@ The following HTTP headers can be used to performing an impersonation request:
 * `Impersonate-User`: The username to act as.
 * `Impersonate-Group`: A group name to act as. Can be provided multiple times to set multiple groups. Optional. Requires "Impersonate-User"
 * `Impersonate-Extra-( extra name )`: A dynamic header used to associate extra fields with the user. Optional. Requires "Impersonate-User". In order to be preserved consistently, `( extra name )` should be lower-case, and any characters which aren't [legal in HTTP header labels](https://tools.ietf.org/html/rfc7230#section-3.2.6) MUST be utf8 and [percent-encoded](https://tools.ietf.org/html/rfc3986#section-2.1).
+* `Impersonate-Uid`: The {{< glossary_tooltip text="UID" term_id="uid" >}} associated with the user being impersonated. Optional. Requires "Impersonate-User".
 
 {{< note >}}
 Prior to 1.11.3 (and 1.10.7, 1.9.11), `( extra name )` could only contain characters which were [legal in HTTP header labels](https://tools.ietf.org/html/rfc7230#section-3.2.6).
 {{< /note >}}
 
+{{< note >}}
+`Impersonate-Uid` is only available in versions 1.22.0 and higher.
+{{< /note >}}
+
 An example set of headers:
```

Comment:
Maybe this is the point to split it into 2 examples. The first example only sets `Impersonate-User:` and `Impersonate-Group:` headers; the second example sets  `Impersonate-User:` and every other header from the proposed example here, _except_ `Impersonate-Group:`

Motivation: make it easier to spot which headers are optional; reduce risk of any readers guessing that `Impersonate-UID:` is mandatory.

(it's fine to merge this as-is, I might propose that tweak separately once this merges).

Reply:
Resolved, I went ahead and split it.